gradio/components/dialogue.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from gradio.components.base import server
+from gradio.components.textbox import Textbox
+from gradio.data_classes import GradioModel, GradioRootModel
+from gradio.events import Events
+
+
+class DialogueLine(GradioModel):
+    speaker: str
+    text: str
+
+
+class DialogueModel(GradioRootModel):
+    root: list[DialogueLine] | str
+
+
+class Dialogue(Textbox):
+    """
+    Creates a dialogue components for users to enter dialogue between speakers.
+
+    Demos: dia_dialogue_demo
+    """
+
+    EVENTS = [
+        Events.change,
+        Events.input,
+        Events.submit,
+    ]
+
+    data_model = DialogueModel
+
+    def __init__(
+        self,
+        value: list[dict[str, str]] | Callable | None = None,
+        *,
+        speakers: list[str] | None = None,
+        formatter: Callable | None = None,
+        emotions: list[str] | None = None,
+        separator: str = " ",
+        label: str | None = "Dialogue",
+        info: str
+        | None = "Type colon (:) in the dialogue line to see the available emotion and intonation tags",
+        placeholder: str | None = "Enter dialogue here...",
+        show_label: bool | None = None,
+        container: bool = True,
+        scale: int | None = None,
+        min_width: int = 160,
+        interactive: bool | None = None,
+        visible: bool = True,
+        elem_id: str | None = None,
+        autofocus: bool = False,
+        autoscroll: bool = True,
+        elem_classes: list[str] | str | None = None,
+        render: bool = True,
+        key: int | str | None = None,
+        max_lines: int | None = None,
+        show_submit_button: bool = True,
+        show_copy_button: bool = True,
+    ):
+        """
+        Parameters:
+            value: Value of the dialogue. It is a list of dictionaries, each containing a 'speaker' key and a 'text' key. If a function is provided, the function will be called each time the app loads to set the initial value of this component.
+            speakers: The different speakers allowed in the dialogue.
+            formatter: A function that formats the dialogue line dictionary, e.g. {"speaker": "Speaker 1", "text": "Hello, how are you?"} into a string, e.g. "Speaker 1: Hello, how are you?".
+            emotions: The different emotions and intonation allowed in the dialogue. Emotions are displayed in an autocomplete menu below the input textbox when the user starts typing `:`. Use the exact emotion name expected by the AI model or inference function.
```

Comment:
as discussed in slack somewhere, consider a more general name for this parameter like `completions`
```suggestion
            emotions: The different emotions and intonation allowed in the dialogue. Emotions are displayed in an autocomplete menu below the input textbox when the user starts typing `:`. Use the exact emotion name expected by the AI model or inference function.
```